homeassistant/components/light/lw12wifi.py

```diff
@@ -0,0 +1,214 @@
+    transition: 10
+    # (Optional) LED Brightness (Value: 0-255)
+    brightness: 10
+    # (Optional) RGB value array as standard color to set.
+    # This option is is ignored when effect is set.
+    rgb: [255, 255, 255]
+"""
+
+import logging
+import time
+
+import voluptuous as vol
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_EFFECT, ATTR_HS_COLOR, ATTR_TRANSITION,
+    Light, PLATFORM_SCHEMA, SUPPORT_BRIGHTNESS, SUPPORT_EFFECT,
+    SUPPORT_COLOR, SUPPORT_TRANSITION
+)
+from homeassistant.const import (
+    CONF_BRIGHTNESS, CONF_EFFECT, CONF_HOST, CONF_NAME, CONF_PORT, CONF_RGB
+)
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util.color as color_util
+
+
+REQUIREMENTS = ['lw12==0.9.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TRANSITION = 'transition'
+
+DEFAULT_NAME = 'LW-12 FC'
+DEFAULT_PORT = 5000
+DEFAULT_RGB = [255, 255, 255]
+DEFAULT_BRIGHTNESS = 255
+DEFAULT_EFFECT = None
+DEFAULT_TRANSITION = 128
+
+DOMAIN = 'lw12wifi'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_RGB, default=DEFAULT_RGB):
+        vol.All(list, vol.Length(min=3, max=3),
+                [vol.All(vol.Coerce(int), vol.Range(min=0, max=255))]),
+    vol.Optional(CONF_BRIGHTNESS, default=DEFAULT_BRIGHTNESS):
+        vol.All(vol.Coerce(int), vol.Range(min=0, max=255)),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_TRANSITION, default=DEFAULT_TRANSITION):
+        vol.All(vol.Coerce(int), vol.Range(min=0, max=255)),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup LW-12 WiFi LED Controller platform."""
+    # Assign configuration variables.
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    rgb = config.get(CONF_RGB)
+    brightness = config.get(CONF_BRIGHTNESS)
+    effect = config.get(CONF_EFFECT)
+    transition = config.get(CONF_TRANSITION)
+    # Add devices
+    add_devices([LW12WiFi(name, host, port, rgb, brightness, effect,
+                          transition)])
+
+
+class LW12WiFi(Light):
+    """LW-12 WiFi LED Controller."""
+
+    def __init__(self, name, host, port, rgb_color, brightness, effect,
+                 transition):
+        """Initialisation of LW-12 WiFi LED Controller.
+
+        Args:
+            name: Friendly name for this platform to use.
+            host: Hostname or IP address of the device.
+            port: Port (Default: 5000) to connect to.
+            rgb_color: Initial color to set after the light turned on.
+            brightness: Brightness of the LEDs to set.
+            effect: If not None, turn on the lights with the selected effect.
+            transition: Speed of the effects.
+        """
+        import lw12
+
+        self._light = lw12.LW12Controller(host, port)
```

Comment:
Move that to `setup_platform` and pass it down to `LW12WiFi()`. This way you don't need `host` and port`.